torch/testing/_internal/distributed/rpc/dist_autograd_test.py

```diff
@@ -2756,8 +2860,8 @@ def test_gradients_synchronizations(self):
                     WrapperModule,
                     args=(layers[rank - 1], rank)
                 ))
+            x = x.to(0)
 
-            x = torch.randn(5000, 2000).to(0)
             # local iteration
             local_model = nn.Sequential(*local_layers)
             local_model(x).sum().backward()
```

Comment:
curious, IIUC, x can be a sparse tensor here and `local_model` contains a sequence of dense layers. This means we can directly pass sparse tensors to dense layers?

Reply:
Yes.

```python
class MyModel(torch.nn.Module):

    def __init__(self):
        super().__init__()
        self.fc = torch.nn.Linear(4, 4)

    def forward(self, x):
        return self.fc(x)

if __name__ == "__main__":
    x = torch.rand(4,4).to_sparse()
    m = MyModel()
    out = m(x)
    print(out)
 ```